src/controllers/signUp.js

```diff
@@ -1,6 +1,66 @@
-exports.get = (req,res) => {
-    res.render("sign-up", {
-        title: "Sign-up", headerFound: false, footerFound: false, asideFound:false, style: ["signUpIn"],
-         javascript: ['sign-up']
+const bcrypt = require('bcryptjs');
+const addUserDB = require('./../model/queries/addUser');
+
+// Vaild Username
+const vaildUsername = userObj => new RegExp('^[a-zA-Z0-9]{5,}$').test(userObj.username);
+
+// Vaild Email
+const vaildEmail = userObj => !new RegExp('^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$')
+  .test(userObj.email);
+
+// Vaild Password
+const vaildPassword = userObj => (new RegExp('^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#\$%\^&\*])(?=.{8,})')
+  .test(userObj.password)
+  || new RegExp('^(((?=.*[a-z])(?=.*[A-Z]))|((?=.*[a-z])(?=.*[0-9]))|((?=.*[A-Z])(?=.*[0-9])))(?=.{6,})')
+    .test(userObj.password));
+
+// Vaild userObject
+const vaildUserObject = userObj => (vaildPassword(userObj) && vaildUsername(userObj) && vaildEmail(userObj));
+
+// hasing password
+const hasingPassword = (password, res, callback) => {
+  bcrypt.genSalt(10, (err, salt) => {
```

Comment:
Please change it to the promise version of the function :
bcrypt.hash(myPlaintextPassword, saltRounds).then(function(hash) {
    // Store hash in your password DB.
});
// Load hash from your password DB.
bcrypt.compare(myPlaintextPassword, hash).then(function(res) {
    // res == true
});
bcrypt.compare(someOtherPlaintextPassword, hash).then(function(res) {
    // res == false
});
https://www.npmjs.com/package/bcrypt

Reply:
i will try that